File: assets/js/models/cards/Property.ts
export class Property {
  public color: string
  public name: string
  public rentValues: number[]
  public value: number

  constructor(params: Partial<Property>) {
    this.color = params.color
    this.name = params.name
    this.rentValues = params.rentValues
    this.value = params.value
  }
}

File: lib/monoboly_deal/deck/property.ex
defmodule MonobolyDeal.Deck.Property do
  alias MonobolyDeal.Deck.Property

  @derive {Jason.Encoder, only: [:name, :value, :color, :rent_values]}
  @enforce_keys [:name, :color, :value, :rent_values]
  defstruct [:name, :color, :value, :rent_values]

  def new(value, color, rent_values) do
    %Property{
      name: :property_card,
      value: value,
      color: color,
      rent_values: rent_values
    }
  end
end
